test/core/ext/filters/event_engine_client_channel_resolver/resolver_fuzzer.cc

```diff
@@ -0,0 +1,265 @@
+              absl::StatusOr<std::vector<ResolvedAddress>> response) mutable {
+            runner->Run(
+                [cb = std::move(cb), response = std::move(response)]() mutable {
+                  cb(response);
+                });
+            return EventEngine::DNSResolver::LookupTaskHandle::kInvalid;
+          };
+      if (!engine_->hostname_responses_.has_value()) {
+        return finish(engine_->lookup_hostname_response_base_case_);
+      }
+      return finish(*engine_->hostname_responses_);
+    }
+    LookupTaskHandle LookupSRV(LookupSRVCallback on_resolve,
+                               absl::string_view /* name */,
+                               Duration /* timeout */) override {
+      auto finish =
+          [cb = std::move(on_resolve), runner = &engine_->runner_](
+              absl::StatusOr<std::vector<SRVRecord>> response) mutable {
+            runner->Run(
+                [cb = std::move(cb), response = std::move(response)]() mutable {
+                  cb(response);
+                });
+            return EventEngine::DNSResolver::LookupTaskHandle::kInvalid;
+          };
+      if (!engine_->srv_responses_.has_value()) {
+        return finish(engine_->lookup_srv_response_base_case_);
+      }
+      return finish(*engine_->srv_responses_);
+    }
+    LookupTaskHandle LookupTXT(LookupTXTCallback on_resolve,
+                               absl::string_view /* name */,
+                               Duration /* timeout */) override {
+      auto finish =
+          [cb = std::move(on_resolve), runner = &engine_->runner_](
+              absl::StatusOr<std::vector<std::string>> response) mutable {
+            runner->Run(
+                [cb = std::move(cb), response = std::move(response)]() mutable {
+                  cb(response);
+                });
+            return EventEngine::DNSResolver::LookupTaskHandle::kInvalid;
+          };
+      if (!engine_->txt_responses_.has_value()) {
+        return finish(engine_->lookup_txt_response_base_case_);
+      }
+      return finish(*engine_->txt_responses_);
+    }
+    bool CancelLookup(LookupTaskHandle /* handle */) override { return false; }
+
+   private:
+    FuzzingResolverEventEngine* engine_;
+  };
+
+  // members
+  FuzzingEventEngine runner_;
+
+  // responses
+  absl::optional<std::vector<EventEngine::ResolvedAddress>> hostname_responses_;
+  absl::optional<std::vector<EventEngine::DNSResolver::SRVRecord>>
+      srv_responses_;
+  absl::optional<std::vector<std::string>> txt_responses_;
+
+  // base cases
+  const std::string txt_valid_config_ =
+      "grpc_config=[{\"serviceConfig\":{\"loadBalancingPolicy\":\"round_"
+      "robin\",\"methodConfig\":[{\"name\":[{\"method\":\"Foo\",\"service\":"
+      "\"SimpleService\"}],\"waitForReady\":true}]}}]";
+  const std::string txt_invalid_config_ = "grpc_config=[{\"foops\":{}}]";
+  const absl::Status lookup_hostname_response_base_case_ =
```

Comment:
Random thought: would it be useful if we fuzz the error status as well? Maybe something like:

Proto:
```
message Error {
  uint16 code = 1;
  string message = 2;
}
```

And then in the code we can probably create the absl::Status with something like (not sure if the casting would work):
```
// clamp error.code() in (0, 16]
status = absl::Status(static_cast<absl::StatusCode>(error.code() % 16 + 1), error.message());
```

A little bit hacky though...

Reply:
That's a good point. There is no contract for a subset of valid statuses from the EventEngine DNS resolver APIs, so this could be valuable.

Reply:
Maybe use the rpc status proto here?

Reply:
(getting error codes outside range is actually valuable as defense against future changes)

Reply:
The rpc status proto would still need to be converted to an absl::Status for use in the resolver API, so I don't see how that's much of a win.

Reply:
It's implemented for now per Yijie's recommendation.

Reply:
Fair. It's probably worth throwing some any's into the status... we do some funky things based on attributes on statuses throughout the stack (we need not to, but we do) and so it might be interesting to trigger them *at some point*